### DISCUSSION TITLE
Add Axis Camera Feed to Website

### DISCUSSION THREAD

**Discussion by mrod5353** [AxisCommunications/discussions#226] [General] [answered]
Does anyone know of a good way to add a secured Axis camera stream to a website?

I am trying to get a stream to autoplay when the page loads without prompting the user for credentials to an angular site with a .NET backend.

I have tried a couple of different approaches, but with no success. Here is what I have tried.

**url in img tag**
`<img id="stream" src="http://00.00.000.00/axis-cgi/mjpg/video.cgi?resolution=640X360" (error)="streamError($event)" alt="Camera Feed" border="0" />`
Works great, but will not work with a camera that requires authentication to be viewed. 

**img tag http request intercept and add basic authentication header to request**
Wasn't able to get it to work. 

**media-stream-library-js (Client side)**
Works, but client prompts for camera credentials. Doesn't seem to have a way to pass credentials instead of prompting in browser.

**request proxy**
I am currently trying to set up a proxy to authenticate and pass back the stream to the browser, but not having much luck.

If anyone has any solutions, guidance, suggestions, tips, pointers or tidbits of advice it would be greatly appreciated.

**Comment by SampsaKaskela** [comment]
If you followed media-stream-player examples where request is made to usergroup.cgi to authenticate then you notice there is no-cors mode. Having this mode causes that you can't send authorization header with request. So you need to do one of the following things:

1. Configure CORS so you don't need no-cors mode. I have made question about this since I have issue too. You can look my setup there for some guidance. (https://github.com/orgs/AxisCommunications/discussions/223)
2. You proxy requests via your own server and add proper headers between and configure CORS there.
3. You have camera located under same host as your client so CORS is no longer needed.

Hope this helps. Wish I had complete answer but hopefully someone is able to answer my question about CORS so you would get answer too.

**Comment by shahnidhi20** [comment]
HI, I need some sample on how to embed the Axis IP camera in the website. We use ASP dot net application. 
Please share the sample if anyone has it. 

We need to use RTSP live streaming protocol

**Reply by DavGarcia** [reply]
@shahnidhi20 You aren't going to be able to use RTSP natively in a browser. You'll want to use HTTP or HTTPS with MJPEG (image tag in HTML) or MP4 (video tag in HTML).

**Comment by shahnidhi20** [comment]
Do you have a sample example on how to integrate the camera in the web page
.

We are seeing delay in the stream of the live view.

**Reply by DavGarcia** [reply, marked as answer]
There are a number of examples here https://github.com/AxisCommunications/media-stream-library-js/tree/main

**Comment by shahnidhi20** [comment]
Thank you. we have implemented it in the same way as it is described in the
samples.

The problem we are facing is the delay on a video stream.
If I open two camera feeds and move the camera for some time. After some
time we see one of the camera live view responds with a delay. We saw
sometimes a delay of 5 sec up to 30 sec.

example: I opened the Axis web page of the camera in the browser and opened
my web page which also shows a live view of the camera. After a couple of
mins, we saw the delay on either our web page or the Axis web page.

If you can share some light on what could be the issue or any settings in
the camera that needs to be done that will be helpful.


Thanks
Nidhi Shah
